networkx/algorithms/isomorphism/matchhelpers.py

```diff
@@ -99,8 +99,8 @@ def match(data1, data2):
     else:
         attrs = list(zip(attr, default)) # Python 3
         def match(data1, data2):
-            values1 = set([data1.get(attr, d) for attr, d in attrs])
-            values2 = set([data2.get(attr, d) for attr, d in attrs])
+            values1 = [data1.get(attr, d) for attr, d in attrs]
+            values2 = [data2.get(attr, d) for attr, d in attrs]
             return values1 == values2
```

Comment:
Might be better as

    return all(data1.get(k, d) == data2.get(k, d) for k, d in zip(attr, default))

so that lists don't need to be created.